docs/how-to-guides/data-basics/5-adding-a-delete-button.md

```diff
@@ -0,0 +1,454 @@
+# Adding a delete button
+
+In this section, we will add a *Delete* feature to our app. Here's a glimpse of what we're going to build:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/delete-button.png)
+
+### Step 1: Add a _Delete_ button
+
+Let's start by creating the `DeletePageButton` component and updating the user interface of our `PagesList` component:
+
+```js
+import { Button } from '@wordpress/components';
+import { decodeEntities } from '@wordpress/html-entities';
+
+const DeletePageButton = () => (
+	<Button variant="primary">
+		Delete
+	</Button>
+)
+
+function PagesList( { hasResolved, pages } ) {
+	if ( ! hasResolved ) {
+		return <Spinner />;
+	}
+	if ( ! pages?.length ) {
+		return <div>No results</div>;
+	}
+
+	return (
+		<table className="wp-list-table widefat fixed striped table-view-list">
+			<thead>
+				<tr>
+					<td>Title</td>
+					<td style={{width: 190}}>Actions</td>
+				</tr>
+			</thead>
+			<tbody>
+				{ pages?.map( ( page ) => (
+					<tr key={page.id}>
+						<td>{ decodeEntities( page.title.rendered ) }</td>
+						<td>
+							<div className="form-buttons">
+								<EditPageButton pageId={ page.id } />
+								<DeletePageButton pageId={ page.id } />
+							</div>
+						</td>
+					</tr>
+				) ) }
+			</tbody>
+		</table>
+	);
+}
+```
+
+This is what the PagesList should look like now:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/delete-button.png)
+
+### Step 2: Wire the button to a delete action
+
+In Gutenberg data, we delete entity records from the WordPress REST API using the `deleteEntityRecord` action. It sends the request, processes the result, and updates the cached data in the Redux state.
+
+Here's how you can try it in your browser's dev tools:
+
+```js
+// We need a valid page ID to call deleteEntityRecord, so let's get the first available one using getEntityRecords.
+const pageId = wp.data.select( 'core' ).getEntityRecords( 'postType', 'page' )[0].id;
+
+// Now let's delete that page:
+const promise = wp.data.dispatch( 'core' ).deleteEntityRecord( 'postType', 'page', pageId );
+
+// promise gets resolved or rejected when the API request succeeds or fails.
+```
+
+Once the REST API request is finished, you will notice one of the pages has disappeared from the list. This is because that list is populated by using `useSelect()` and the `select( coreDataStore ).getEntityRecords( 'postType', 'page' )` selector. Anytime the underlying data changes, the list gets re-rendered with fresh data. That's pretty convenient!
```

Comment:
```suggestion
Once the REST API request is finished, you will notice one of the pages has disappeared from the list. This is because that list is populated by the `useSelect()` hook and the `select( coreDataStore ).getEntityRecords( 'postType', 'page' )` selector. Anytime the underlying data changes, the list gets re-rendered with fresh data. That's pretty convenient!
```